### ISSUE TITLE
Confusing findings for dictionary constructors

### ISSUE THREAD

**Issue by tjs-intel** [pschanely/CrossHair#257]
**Expected vs actual behavior**
I expect all the dictionary constructors in the example to have indistinguishable behavior. I found that the `dict` constructor with a generator of tuples yields false positives.

**To Reproduce**
```
> crosshair check example.py
***/example.py:31: error: false when calling test_func_broken({'': ''}) (which returns {})
> cat example.py
from typing import Mapping, Hashable, TypeVar

V = TypeVar("V")

def test_func_valid_0(d: Mapping[str, V]) -> dict[str, V]:
  '''
  post: d == __return__
  '''
  return {k: v for k, v in d.items()}

def test_func_valid_1(d: Mapping[str, V]) -> dict[str, V]:
  '''
  post: d == __return__
  '''
  return dict(d)

def test_func_valid_2(d: Mapping[str, V]) -> dict[str, V]:
  '''
  post: d == __return__
  '''
  return dict(d.items())

def test_func_valid_3(d: Mapping[str, V]) -> dict[str, V]:
  '''
  post: d == __return__
  '''
  return dict([(k, v) for k, v in d.items()])

def test_func_broken(d: Mapping[str, V]) -> dict[str, V]:
  '''
  post: d == __return__
  '''
  return dict((k, v) for k, v in d.items())
```

**Comment by tjs-intel** [comment]
@pschanely

**Comment by pschanely** [comment]
Interesting! Thank you for the detailed report. I will investigate today and get back to you!

**Comment by pschanely** [comment]
I can reproduce - I imagine this may have been a regression since v0.0.50.
The fix is nearly ready and I'll be able to release it shortly.

**Comment by pschanely** [comment]
This should be fixed since [v0.0.53](https://crosshair.readthedocs.io/en/latest/changelog.html#version-0-0-53)! I'd love it if you could try again and see how it goes.

**Comment by tjs-intel** [comment]
Seems to work now! Thanks for the quick fix!
